libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2724,6 +2780,16 @@ string YulUtilFunctions::storageSetToZeroFunction(Type const& _type)
 			("functionName", functionName)
 			("clearArray", clearStorageArrayFunction(dynamic_cast<ArrayType const&>(_type)))
 			.render();
+		else if (_type.category() == Type::Category::Struct)
+			return Whiskers(R"(
+				function <functionName>(slot, offset) {
+					if iszero(eq(offset, 0)) { invalid() }
```

Comment:
Hm... I'm wondering whether we should have this check or not (even though I asked for it myself at some point IIRC ;-))... If we want to keep it, we should probably add it to the ``Array`` case above as well for consistency.

Ideally the optimizer would generally remove these checks again, but e.g. in the use in ``storageArrayPopFunction`` we'd probably need to trust that the ``<indexAccess>`` is inlined for that to work...

But well, generally I think we said safety first with new codegen and we can tweak the optimizer or manually optimize the code later. So maybe we should keep it (and add the same check to the array case)?

Reply:
For me it is better to leave this check. Hopefully optimizer will be able to optimize it out, so it shouldn't hurt, and we will be on the safe side.
Adding the same to array case would make sense in that case.

Reply:
Added same check for array case